src/devices/transitions/transitionHandler.ts

```diff
@@ -0,0 +1,178 @@
+import { TSRTransitionOptions } from '../../types/src/casparcg'
+import * as _ from 'underscore'
+import { Animator, LinearMovement, PhysicalAcceleration } from './animate'
+
+
+
+interface TransitionHandler {
+	values: number[]
+	target: number[]
+	groups: string[]
+
+	calculatingGroups: {[groupId: string]: {
+		animator: Animator
+	}}
+
+	activeIterator: NodeJS.Timeout | null
+	lastUpdate: number
+	updateCallback?: (newValues: number[]) => void
+}
+export class InternalTransitionHandler {
+	private _transitions: {[identifier: string]: TransitionHandler} = {}
+
+	public terminate () {
+		// clearInterval(this._interval)
+		_.each(this._transitions, (_transition, identifier) => {
+			this.clearTransition(identifier)
+		})
+	}
+	public getIdentifiers () {
+		return Object.keys(this._transitions)
+	}
+
+	public clearTransition (identifier: string) {
+		const t = this._transitions[identifier]
+		if (t) {
+			this._stopTransition(t)
+			delete this._transitions[identifier]
+		}
+	}
+	public snapTransition (
```

Comment:
Would this method name be more self-explanatory if it was called `stopAndSnapTransition`?